### PR TITLE
[5.8] Update to pda/pheanstalk ^4.0

### PR DESCRIPTION
**Pull request by crynobone**
Release note: https://github.com/pheanstalk/pheanstalk/releases/tag/v4.0.0

Biggest improvement would be better handling of closed/failed connection to beanstalkd which in previous version you need to wait until exception is thrown before the worker get restarted or you have to do a schedule task to restart the workers on every interval (mine was 24 hours).

Signed-off-by: Mior Muhammad Zaki <crynobone@gmail.com>

### CODE REVIEW

File: src/Illuminate/Queue/Connectors/BeanstalkdConnector.php [hunk 1]
```diff
@@ -32,11 +32,10 @@ public function connect(array $config)
      */
     protected function pheanstalk(array $config)
     {
-        return new Pheanstalk(
+        return Pheanstalk::connect(
```
Comment (driesvints) [inline src/Illuminate/Queue/Connectors/BeanstalkdConnector.php]: @crynobone I've been looking at the code here and I think something's off. There doesn't seems to be a `connect` method in v4. You've added the `create` method which is the correct one I believe in https://github.com/laravel/framework/pull/27622

Comment (driesvints) [reply src/Illuminate/Queue/Connectors/BeanstalkdConnector.php]: If you fix this can you perhaps send in a test?

Comment (crynobone) [reply src/Illuminate/Queue/Connectors/BeanstalkdConnector.php]: Sorry, yes `create()` is the correct method. 

To add a test for this we need to include beanstalk on travis, okay on that?

Comment (driesvints) [reply src/Illuminate/Queue/Connectors/BeanstalkdConnector.php]: No that's ok. I've pushed a commit with the correct method. Thanks!

**Pull request merged**